lib/snowplow_ruby_duid/cookie.rb

```diff
@@ -11,10 +11,12 @@ class Cookie
     # See: https://github.com/rails/rails/blob/b1124a2ac88778c0feb0157ac09367cbd204bf01/actionpack/lib/action_dispatch/middleware/cookies.rb#L214
     DOMAIN_REGEXP          = /[^.]*\.([^.]*|..\...|...\...)$/.freeze
 
-    def initialize(host, domain_userid, request_created_at)
+    def initialize(host, domain_userid, request_created_at, options = {})
       @host               = host
       @domain_userid      = domain_userid
       @request_created_at = request_created_at
+      @secure             = options.fetch(:secure)
```

Comment:
Is it meant to raise exception if no option value is provided? Should it rather fall into something sensible? Like, `false` and, alternatively, log message if value is not set explicitly?

Reply:
Yeah, it should fail, as this attribute is required.

But in reality this object is private in a sense, and controlled by the `helper.rb` which passes the `Configuration.secure` option, that has a default.

Reply:
👍